### DISCUSSION TITLE
Chalet Build Tool

### DISCUSSION THREAD

**Discussion by dimitre** [openframeworks/openFrameworks#7123] [Ideas]
Have you used this build system? I think it is very promising for multi platform projects like OF.
https://www.chalet-work.space

if you want to try it out on macOS you can install by this homebrew formula
```
brew install --cask https://raw.githubusercontent.com/dimitre/chalet_homebrew/main/chalet.rb
```

and grab the configuration here:
https://gist.github.com/dimitre/4339fda7bef0d7ff373c2da2f5d2c25f
with the file chalet.json on the root OF folder you can run using the command

```
chalet buildrun
```

it is great for different platforms customization, it accept glob paths, so it is easy to add lots of sources, exclude others and all the fiddling needed to make things run. this first attempt doesn't work with addons

but I liked a lot the way it works. in order to run you don't need to copy plist, icons, everything. it will just compile and run, so it is fast
there is a lot of configurations possible when deploying an app, including a dmg installer, custom icons, everything to streamline distribution.

Maybe it is a nice alternative to the whole build system including apothecary, makefiles and project generator.

speed: build a clean OF project (ofBoxExample) using
make: 42.58s
Chalet: 20.18s

**Comment by danoli3** [comment]
I still would recommend cmake it made android like 10s
